dynamodb-stream-reader/src/main/java/elasticsearch/ElasticSearchRestClient.java

```diff
@@ -0,0 +1,124 @@
+
+import static no.unit.nva.dynamodb.DynamoDBStreamHandler.ELASTICSEARCH_ENDPOINT_INDEX_KEY;
+
+public class ElasticSearchRestClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(ElasticSearchRestClient.class);
+
+    private static final String ELASTICSEARCH_ENDPOINT_OPERATION = "_doc";
+    public static final String INITIAL_LOG_MESSAGE = "using Elasticsearch endpoint {} {} and index {}";
+    public static final String APPLICATION_JSON = "application/json";
+
+    public static final String UPSERTING_LOG_MESSAGE = "Upserting search index  with values {}";
+    public static final String DELETE_LOG_MESSAGE = "Deleting from search API publication with identifier: {}";
+    public static final String ELASTICSEARCH_ENDPOINT_ADDRESS_KEY = "ELASTICSEARCH_ENDPOINT_ADDRESS";
+    public static final String ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY = "ELASTICSEARCH_ENDPOINT_API_SCHEME";
+    public static final String ELASTICSEARCH_ENDPOINT_URI_TEMPLATE = "%s://%s/%s/%s/%s";
+    public static final String POSTING_TO_ENDPOINT_LOG_MESSAGE = "POSTing {} to endpoint {}";
+    public static final String MISSING_IN_ENVIRONMENT_ERROR = "Missing '%s' in environment";
+
+    private final HttpClient client;
+    private final String elasticSearchEndpointAddress;
+    private final String elasticSearchEndpointIndex;
+    private final String elasticSearchEndpointScheme;
+
+    /**
+     * Creates a new ElasticSearchRestClient.
+     *
+     * @param httpClient Client to speak http
+     * @param environment Environment with properties
+     */
+    public ElasticSearchRestClient(HttpClient httpClient, Environment environment) {
+        client = httpClient;
+        elasticSearchEndpointAddress = environment.readEnv(ELASTICSEARCH_ENDPOINT_ADDRESS_KEY);
+        elasticSearchEndpointIndex = environment.readEnv(ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        elasticSearchEndpointScheme = environment.readEnv(ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY);
+
+        logger.info(INITIAL_LOG_MESSAGE,
+                elasticSearchEndpointScheme, elasticSearchEndpointAddress, elasticSearchEndpointIndex);
+    }
+
+    /**
+     * Adds or insert a document to an elasticsearch index.
+     * @param document the document to be inserted
+     * @throws URISyntaxException thrown when uri is misconfigured
+     * @throws IOException thrown hen service i not available
+     * @throws InterruptedException thrown when service i interrupted
+     */
+    public void addDocumentToIndex(ElasticSearchIndexDocument document)
+            throws URISyntaxException, IOException, InterruptedException {
+        logger.debug(UPSERTING_LOG_MESSAGE, document);
+
+        HttpRequest request = createHttpRequest(document);
+
+        HttpResponse<String> response = doSend(request);
+        logger.debug(response.body());
+    }
+
+    private HttpRequest createHttpRequest(ElasticSearchIndexDocument document) throws
+            JsonProcessingException, URISyntaxException {
```

Comment:
it does not throw JsonProcessingException

Reply:
It doesn't anymore..